# Prueba de Chi-Cuadrado
Para realizar la Prueba de Chi-Cuadrado, necesitaremos 20 números aleatorios. Podemos dividir estos números en intervalos y luego comparar la frecuencia observada en cada intervalo con la frecuencia esperada si los números fueran uniformemente distribuidos.

## Generación de 20 Números Aleatorios

Vamos a generar 20 números aleatorios en el rango [0, 1] y luego los dividiremos en 4 intervalos iguales para aplicar la Prueba de Chi-Cuadrado.

In [ ]:
# Importando la librería necesaria
import numpy as np

# Generando 20 números aleatorios en el rango [0, 1]
random_numbers = np.random.rand(20)

# Creando 4 intervalos iguales
intervals = [0.25, 0.5, 0.75, 1.0]

# Contando cuántos números caen en cada intervalo
counts = [0] * 4
for number in random_numbers:
    for i, interval in enumerate(intervals):
        if number <= interval:
            counts[i] += 1
            break

random_numbers, counts

(array([0.53973366, 0.28978919, 0.9481132 , 0.37118651, 0.34448993,
        0.9881888 , 0.18385342, 0.77256748, 0.71668274, 0.83008225,
        0.45712711, 0.8423145 , 0.42867145, 0.39704351, 0.07758309,
        0.70023575, 0.64859747, 0.60339392, 0.46452417, 0.52118346]),
 [2, 7, 6, 5])

Hemos generado 20 números aleatorios y los hemos dividido en 4 intervalos iguales. Las cuentas en cada intervalo son las siguientes:

- Intervalo 1 (0.00 - 0.25): 2 números
- Intervalo 2 (0.25 - 0.50): 7 números
- Intervalo 3 (0.50 - 0.75): 6 números
- Intervalo 4 (0.75 - 1.00): 5 números

## Aplicación de la Prueba de Chi-Cuadrado

La Prueba de Chi-Cuadrado evalúa si las frecuencias observadas en estos intervalos difieren significativamente de lo que esperaríamos si los números fueran uniformemente distribuidos. La fórmula para calcular el estadístico $x^2$ es:

$$x^2=\sum \dfrac{(O_i-E_i)^2}{E_i}$$

Donde

- $O_i$ es la frecuencia observada en el intervalo $i$
- $E_i$ es la frecuencia esperada en el intervalo $i$ (en este caso,$E_i=5$ para todos los intervalos, ya que esperamos una distribución uniforme)

> Nota: $E_i = \dfrac{20}{4} = 5$

Calculando el valor de $x^2$ es:

In [ ]:
# Definiendo la frecuencia esperada en cada intervalo (uniforme)
expected_counts = [5] * 4

# Calculando el valor de Chi-Cuadrado
chi_squared = sum(((o - e) ** 2) / e for o, e in zip(counts, expected_counts))
chi_squared

2.8000000000000003

## Cálculo del Estadístico $x^2$

Las frecuencias observadas (O) y esperadas (E) son las siguientes:

- Intervalo 1: $O=2,E=5$
- Intervalo 2: $O=7,E=5$
- Intervalo 3: $O=6,E=5$
- Intervalo 4: $O=5,E=5$

Aplicando la fórmula:

El valor calculado para el estadístico $x^2$ es: 2.8

$$ x^2 = \dfrac{(2-5)^2}{5}+\dfrac{(7-5)^2}{5}+\dfrac{(6-5)^2}{5}+\dfrac{(5-5)^2}{5} $$

$$ = \dfrac{9}{5}+\dfrac{4}{5}+\dfrac{1}{5}+\dfrac{0}{5} $$

$$ = \dfrac{14}{5} $$

$$ = 2.8 $$


# Conclusión

Al comparar este valor con el valor crítico de la distribución $x^2$ con 3 grados de libertad (4 intervalos menos 1) y a un nivel de significancia de 0.05, el valor crítico es aproximadamente 7.815.

Dado que nuestro valor calculado de $x^2$ es menor que el valor crítico, no rechazamos la hipótesis nula de que los números son uniformemente distribuidos.

Por lo tanto, la secuencia de 20 números aleatorios proporcionados **pasa** la Prueba de Chi-Cuadrado para la uniformidad.